actions/scoring.js

```diff
@@ -48,19 +49,30 @@ async function getAllTokenBalances(address, chainId) {
     return balances;
   } catch (e) {
     console.log("Error: " + e);
-    toast.error("Couldn't fetch Covalent balances. See console.");
+    // toast.error("Couldn't fetch Covalent balances. See console.");
   }
 }
 
-async function getUSDBalance(address) {
-  let balanceArray = await getAllTokenBalances(address);
+async function getUSDBalance(tokenList) {
+  let balanceArray = tokenList;
   let result = 0;
   for (let obj in balanceArray) {
     result += balanceArray[obj].USD;
   }
   return result;
```

Comment:
Could be reduced as:

```js
return balanceArray.reduce((a, b) => a.USD + b.USD);
```

Reply:
returns NaN, do you know why ?

Reply:
Oh yeah it might need nome refactoring, depending on if it's an object or not.
I didn't write the whole working code.
No need to change it right now anyways.